core/src/audio_key.rs

```diff
@@ -73,14 +77,13 @@ impl AudioKeyManager {
 
 pub struct AudioKeyFuture<T>(oneshot::Receiver<Result<T, AudioKeyError>>);
```

Comment:
I think it might be possible to remove the AudioKeyFuture, make the `request` fn async which currently returns a `AudioKeyFuture` and then directly await the receiver.

Reply:
Yep, done! :-)